storage/aws/aws.go

```diff
@@ -565,140 +598,151 @@ func (s *spannerSequencer) assignEntries(ctx context.Context, entries []*tessera
-		// Figure out which is the starting index of sequenced entries to start consuming from.
-		row, err := txn.ReadRowWithOptions(ctx, "IntCoord", spanner.Key{0}, []string{"seq"}, &spanner.ReadOptions{LockHint: spannerpb.ReadRequest_LOCK_HINT_EXCLUSIVE})
-		if err != nil {
-			return err
-		}
-		var fromSeq int64 // Spanner doesn't support uint64
-		if err := row.Column(0, &fromSeq); err != nil {
-			return fmt.Errorf("failed to read integration coordination info: %v", err)
-		}
-		klog.V(1).Infof("Consuming from %d", fromSeq)
-
-		// Now read the sequenced starting at the index we got above.
-		rows := txn.ReadWithOptions(ctx, "Seq",
-			spanner.KeyRange{Start: spanner.Key{0, fromSeq}, End: spanner.Key{0, fromSeq + int64(limit)}},
-			[]string{"seq", "v"},
-			&spanner.ReadOptions{LockHint: spannerpb.ReadRequest_LOCK_HINT_EXCLUSIVE})
-		defer rows.Stop()
-
-		seqsConsumed := []int64{}
-		entries := make([]storage.SequencedEntry, 0, limit)
-		orderCheck := fromSeq
-		for {
-			row, err := rows.Next()
-			if row == nil || err == iterator.Done {
-				break
+func (s *mysqlSequencer) consumeEntries(ctx context.Context, limit uint64, f consumeFunc, forceUpdate bool) (bool, error) {
+	tx, err := s.dbPool.BeginTx(ctx, nil)
+	if err != nil {
+		return false, fmt.Errorf("failed to begin Tx: %v", err)
+	}
+	defer func() {
+		if tx != nil {
+			if err := tx.Rollback(); err != nil {
+				klog.Errorf("failed to rollback Tx: %v", err)
 			}
+		}
+	}()
 
-			var vGob []byte
-			var seq int64 // spanner doesn't have uint64
-			if err := row.Columns(&seq, &vGob); err != nil {
-				return fmt.Errorf("failed to scan seq row: %v", err)
-			}
+	// Figure out which is the starting index of sequenced entries to start consuming from.
+	row := tx.QueryRowContext(ctx, "SELECT seq FROM IntCoord WHERE id = ? FOR UPDATE", 0)
+	var fromSeq uint64
+	if err := row.Scan(&fromSeq); err == sql.ErrNoRows {
+		return false, nil
+	} else if err != nil {
+		return false, fmt.Errorf("failed to read IntCoord: %v", err)
+	}
+	klog.V(1).Infof("Consuming from %d", fromSeq)
 
-			if orderCheck != seq {
-				return fmt.Errorf("integrity fail - expected seq %d, but found %d", orderCheck, seq)
-			}
+	// Now read the sequenced starting at the index we got above.
+	rows, err := tx.QueryContext(ctx, "SELECT seq, v FROM Seq WHERE id = ? AND seq >= ? ORDER BY seq LIMIT ? FOR UPDATE", 0, fromSeq, limit)
+	if err != nil {
+		return false, fmt.Errorf("failed to read Seq: %v", err)
+	}
+	defer rows.Close()
 
-			g := gob.NewDecoder(bytes.NewReader(vGob))
-			b := []storage.SequencedEntry{}
-			if err := g.Decode(&b); err != nil {
-				return fmt.Errorf("failed to deserialise v: %v", err)
-			}
-			entries = append(entries, b...)
-			seqsConsumed = append(seqsConsumed, seq)
-			orderCheck += int64(len(b))
-		}
-		if len(seqsConsumed) == 0 && !forceUpdate {
-			klog.V(1).Info("Found no rows to sequence")
-			return nil
-		}
+	seqsConsumed := []any{}
```

Comment:
`[]any` rather than `[]uint64` because of something to do with passing these back as SQL args later?
Might be worth a comment here about the slightly unusual type...